Sources/_StringProcessing/Unicode/ASCII.swift

```diff
@@ -99,8 +102,8 @@ extension String {
     }
 
     var next = utf8.index(after: idx)
-    if next == utf8.endIndex {
-      assert(self[idx].isASCII)
+    if next == end {
+      // assert(self[idx].isASCII)
```

Comment:
What about these assertions?

Reply:
It's no longer valid to assert here, since a substring can have a mid-character endpoint that makes the result of `self[idx].isASCII` different from `self[..<end][idx].isASCII`. For example, a substring like this:

```swift
let s = "e\u{301}"
let s1 = Substring(s.unicodeScalars.dropLast())
// s.first!.isASCII == false
// s1.first!.isASCII == true
```